packages/@okta/vuepress-site/docs/reference/architecture-center/directory-coexistence/lab-tutorial-3-generic-database/index.md

```diff
@@ -0,0 +1,316 @@
+
+   </div>
+
+## Import users into Universal Directory with the Users API
+
+With the basic environment set up, you can:
+
+* [Sign in to your Okta account](#sign-in-to-your-okta-account)
+* [Run the import script](#run-the-import-script)
+* [Check the users have imported correctly](#check-the-users-have-imported-correctly)
+
+### Sign in to your Okta account
+
+Stop the currently running containers and sign in to your Okta account.
+
+1. In the `okta-reference-coexistence-db-example` directory, stop the currently running containers with <kbd>Ctrl</kbd> + <kbd>C</kbd>.
+1. Run `okta login`.
+
+   ```bash
+   okta login
+   ```
+
+   If you are already logged into Okta, a prompt similar to the following is returned. The current sign-in configuration is based on the Okta Org URL and API token you provided at your previous okta login command. If you want to use the existing configuration, answer N and skip steps a and b. Otherwise, answer Y and continue to steps a and b.
+
+   ```txt
+   An existing Okta Organization (https://dev-133337.okta.com) was found in C:\mydirectory\.okta\okta.yaml
+   Overwrite configuration file? [Y/n]
+   ```
+
+   a. Enter your `${OKTA_DOMAIN}`. If you do not know your `${OKTA_DOMAIN}`, see [Values and Variables](/docs/reference/architecture-center/directory-coexistence/lab-prerequisites/#values-and-variables).
+
+   b. Follow the instructions in [Create an API token](/docs/guides/create-an-api-token) to create a token. Once you have the token, enter it at the Okta API token prompt.
+
+### Run the import script
+
+Check out the script to import the users and check they have been imported correctly in Universal Directory
+
+1. Check out the main branch of the `https://github.com/oktadev/okta-reference-coexistence-db-example` repository. The main branch contains a migration script and updates to code in the project to replace the database authentication with OpenID Connect (OIDC).
+
+   ```bash
+   git checkout main
+   ```
+
+   After a successful checkout, the `okta-reference-coexistence-db-example` directory contains the `pw-import-hook` subdirectory, the `migrate-user.sh` file, and other files and code changes.
+
+1. Restart the containers:
+
+   ```bash
+   docker compose start
+   ```
+
+1. Run the okta-reference-coexistence-db-example/migrate-users.sh script.
+
+   ```bash
+   ./migrate-users.sh
+   ```
+
+This script uses a database query to the database server to get the user profiles, and the [Okta Users API](/docs/reference/api/users/) to import those users into Okta. For this example, three users are defined in `okta-reference-coexistence-db-example/blob/main/sql/03-users.sql`.
+
+The script imports the two users whose passwords are hashed with a hash that Okta supports: `Admin Istrator` and `User One`. The first time the users sign in, their passwords are rehashed and the hashed passwords are git-bstored in Okta Universal Directory using the `POST https://${yourOktaDomain}/api/v1/user`s` Okta API request.
```

Comment:
```suggestion
The script imports the two users whose passwords are hashed with a hash that Okta supports: `Admin Istrator` and `User One`. The first time the users sign in, their passwords are rehashed and the hashed passwords are git-bstored in Okta Universal Directory using the `POST https://${yourOktaDomain}/api/v1/users` Okta API request.
```

Reply:
You may also want to link to this section in Users API: https://developer.okta.com/docs/reference/api/users/#create-user-with-imported-hashed-password

Reply:
Excellent link. Thank you